README.md

```diff
@@ -0,0 +1,184 @@
+
+A package for generating a flutter `ThemeData` from an exported design token json that is used e.g. [in this Figma plugin](https://tokens.studio/). It rather aims to create a flutter `Theme` 
+based on Material 3 than creating a lot of custom widgets for each token.
+
+## Features
+
+Right now we support generating/parsing the following:
+- `ColorScheme`
+- `TextStyle`
+  - Font family mapping (from Figma font name to flutter name)
+  - Font weight
+  - Line height
+  - Font size
+  - Letter spacing
+  - Text decoration
+- Theme extension
+  - Custom colors
+- Exposing theming and extensions via a `BuildContext` extension
+
+## Getting started
+
+1. Add a `.json` file containing the tokens describing your app design.
+2. Add a `tokenbuilder.yaml` configuration file in your `lib` directory
+   1. Add the path leading to the json to the configuration file 
+   2. Map Figma font names to flutter font family names in configuration file
+3. Add this builder to `build.yaml`
+```yaml
+targets:
+  $default:
+    builders:
+      design_tokens_builder:design_tokens_builder:
+        enabled: true
+```
+
+## Usage
+
+Before you can use the tokens you have to start the build runner by executing `flutter pub run build_runner build`.
+After that just make sure to use one of the `GeneratedTokenSet` for the selected `Brightness` in your `ThemeWidget`.
+```dart
+return Theme(
+  data: GeneratedTokenSet.general.data.dark,
+  child: Container(),
+);
+```
+By listening on Brightness changes you could then easily switch between themes. You could also easily change the token set. 
+
+This package also exposes the generated theme extensions by building a `BuildContext` extension. You can use the shortcut like this `context.yourExtension`. We also provide shortcuts for theme related properties like `context.colorScheme` and `context.textTheme`.
+
+## Additional information
+
+### Multi-theming and dark and light mode
+The package can generate multiple themes based on multiple [token sets](https://docs.tokens.studio/themes/token-sets) in
+your token data json. Every set will be available through a `GeneratedTokenSet`. This enum contains all set pairs and 
+`ThemeData` for light and dark themes. In order to specify the brightness of your set use Light/Dark as a suffix in 
+design tokens (e.g. yourSetLight, yourSetDark). You also might have a global or core set with universal tokens that do 
+not change based on the selected brightness. In that case make sure to call this set `global`. It will be recognised by 
+the builder and ignored when creating `GeneratedTokenSets`.
+
+### Generation of flutter `ThemeData`
+The package allows for easy and simple generation of flutters `ThemeData`. For doing so, you have to have a certain 
+structure for your design tokens. Since this is a system token related to material 3 the tokens used to generate flutters
+`ThemeData` should always start with `sys`. If you then e.g. want to set the primary color in flutters `ColorScheme` you

```

Comment:
Any change the structure and the sys rule can be demonstrated with a code snippet or a screenshot?